### PR TITLE
Music list view model

### PR DESCRIPTION
**Pull request by IamHaddad**
music list view model added

### CODE REVIEW

File: app/src/main/java/com/worldsnas/starplayer/MainActivity.kt [hunk 1]
```diff
@@ -12,9 +18,77 @@ class MainActivity : AppCompatActivity() {
 
     private lateinit var binding: ActivityMainBinding
 
+    private val MY_PERMISSIONS_REQUEST_READ_EXTERNAL_STORAGE = 100
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
 
+        checkPermission()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/MainActivity.kt]: ma ba mainActivity kari nadarim. chera permission inja check mishe?

File: app/src/main/java/com/worldsnas/starplayer/MainActivity.kt [hunk 2]
```diff
@@ -12,9 +18,77 @@ class MainActivity : AppCompatActivity() {
...
+                )
+
+                // MY_PERMISSIONS_REQUEST_READ_CONTACTS is an
+                // app-defined int constant. The callback method gets the
+                // result of the request.
+            }
+        } else {
+
+            // Permission has already been granted
+            Toast.makeText(this, "Already Granted", Toast.LENGTH_SHORT).show()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/MainActivity.kt]: chera vaghti permission dade shode hich kari anjam nemishe?

File: app/src/main/java/com/worldsnas/starplayer/MainActivity.kt [hunk 3]
```diff
@@ -12,9 +18,77 @@ class MainActivity : AppCompatActivity() {
...
+                    this,
+                    Manifest.permission.READ_EXTERNAL_STORAGE
+                )
+            ) {
+                // Show an explanation to the user *asynchronously* -- don't block
+                // this thread waiting for the user's response! After the user
+                // sees the explanation, try again to request the permission.
+            } else {
+                // No explanation needed, we can request the permission.
+                ActivityCompat.requestPermissions(
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/MainActivity.kt]: age man bodam ye extension mineveshtam! vali esrari nist

File: app/src/main/java/com/worldsnas/starplayer/MainActivity.kt [hunk 4]
```diff
@@ -12,9 +18,77 @@ class MainActivity : AppCompatActivity() {
...
+                    this,
+                    arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE),
+                    MY_PERMISSIONS_REQUEST_READ_EXTERNAL_STORAGE
+                )
+
+                // MY_PERMISSIONS_REQUEST_READ_CONTACTS is an
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/MainActivity.kt]: comment haro copy kardid?
benazaram comment haro hamaro pak konid
code engahd vazeh hast ke bege dare che mikone
age nist function seda konin va esme function ro monaseb bezarid

File: app/src/main/java/com/worldsnas/starplayer/MainActivity.kt [hunk 5]
```diff
@@ -12,9 +18,77 @@ class MainActivity : AppCompatActivity() {
...
+            Toast.makeText(this, "Already Granted", Toast.LENGTH_SHORT).show()
+        }
+
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<String>, grantResults: IntArray
+    ) {
+        when (requestCode) {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/MainActivity.kt]: `super` ro seda nazadid

File: app/src/main/java/com/worldsnas/starplayer/MainActivity.kt [hunk 6]
```diff
@@ -12,9 +18,77 @@ class MainActivity : AppCompatActivity() {
...
+            MY_PERMISSIONS_REQUEST_READ_EXTERNAL_STORAGE -> {
+                // If request is cancelled, the result arrays are empty.
+                if ((grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED)) {
+                    // permission was granted, yay! Do the
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/MainActivity.kt]: chera hich kari anjam nashode?

File: app/src/main/java/com/worldsnas/starplayer/MainActivity.kt [hunk 7]
```diff
@@ -12,9 +18,77 @@ class MainActivity : AppCompatActivity() {
...
+                    // functionality that depends on this permission.
+                    Toast.makeText(this, "Storage Permission Needed", Toast.LENGTH_SHORT).show()
+                }
+                return
+            }
+            // Add other 'when' lines to check for other
+            // permissions this app might request.
+            else -> {
+                // Ignore all other requests.
+                Toast.makeText(this, "Did Not Granted", Toast.LENGTH_SHORT).show()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/MainActivity.kt]: in toast eshtebahe

File: app/src/main/java/com/worldsnas/starplayer/di/LocalMusicModule.kt [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+import android.app.Application
+import android.content.ContentResolver
+import android.content.Context
+import com.worldsnas.starplayer.App
+import dagger.Module
+import dagger.Provides
+
+@Module
+class LocalMusicModule {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/di/LocalMusicModule.kt]: esme monasebi nist benazaram!

hamchenin pishnahad mikonam to kotlin baraye module ha az `object` estefade konin
va roye function ha `@Jvmstatic` bezanid.

File: app/src/main/java/com/worldsnas/starplayer/di/MusicListComponent.kt [hunk 9]
```diff
@@ -14,5 +18,7 @@ interface MusicListComponent {
         fun build(): MusicListComponent
 
         fun appComponent(appComponent: AppComponent): Builder
+        @BindsInstance
+        fun contextProvider(context: Context): Builder
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/di/MusicListComponent.kt]: in context leak mishe.
benazaram behtare context Application ro az `AppComponent` begirid va be contentResolver bedid

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+import kotlinx.coroutines.*
+import javax.inject.Inject
+
+class MusicListViewModel @Inject constructor(private val localMusicProviderImpl: LocalMusicProviderImpl) :
+    ViewModel() {
+
+    private val postMusicList = MutableLiveData<List<Music>>()
+    private val viewModelJob = SupervisorJob()
+
+    private val uiScope = CoroutineScope(Dispatchers.Main + viewModelJob)
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: lotfan az ketabkhone `ViewModel-ktx` estefade konin va khodeton nasazid

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.worldsnas.starplayer.model.LocalMusicProviderImpl
+import com.worldsnas.starplayer.model.Music
+import kotlinx.coroutines.*
+import javax.inject.Inject
+
+class MusicListViewModel @Inject constructor(private val localMusicProviderImpl: LocalMusicProviderImpl) :
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: lotfan `LocalMusicProvider` ro inject konin

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+    private val viewModelJob = SupervisorJob()
+
+    private val uiScope = CoroutineScope(Dispatchers.Main + viewModelJob)
+
+    fun getAllMusic(): MutableLiveData<List<Music>> {
+        uiScope.launch {
+            val musicsList = localMusicProviderImpl.getAllMusic()
+
+            postMusicList.postValue(musicsList)
+        }
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: function joda bashe behtare

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt [hunk 13]
```diff
@@ -1,20 +1,27 @@
...
 import com.worldsnas.starplayer.di.MusicListComponent
 import com.worldsnas.starplayer.model.Music
+import com.worldsnas.starplayer.view.ViewModelFactory
+import javax.inject.Inject
 
 class MusicsListFragment : Fragment() {
 
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory
+    private val musicListViewModel by lazy {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: az property delegate ha ye jadid ketabkhone ViewModel estefade konid

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt [hunk 14]
```diff
@@ -32,14 +39,11 @@ class MusicsListFragment : Fragment() {
...
-        musics.add(Music(0, "Test", "Ali", "Single", "Pop","somewhere"))
-        musics.add(Music(0, "Test", "Ali", "Single", "Pop","somewhere"))
-        musics.add(Music(0, "Test", "Ali", "Single", "Pop","somewhere"))
-        musics.add(Music(0, "Test", "Ali", "Single", "Pop","somewhere"))
-        musics.add(Music(0, "Test", "Ali", "Single", "Pop","somewhere"))
-        musics.add(Music(0, "Test", "Ali", "Single", "Pop","somewhere"))
-        binding.recyclerview.adapter = MusicsListAdapter(musics)
+        val musicObserver = Observer<List<Music>> {
+
+            binding.recyclerview.adapter = MusicsListAdapter(it as ArrayList<Music>)
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: in kare dorosti nist
shoma bayad data dakhele adapter ro beroz konid na inke khode adapter ro avaz konid

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
good job
small improvements would be great

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt (hunk 13 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: age emkanesh hast ino bishtar tozih bedid, nmidonm chi hastand

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt (hunk 10 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: viewModelScope mazoretone?

File: app/src/main/java/com/worldsnas/starplayer/di/LocalMusicModule.kt (hunk 8 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/di/LocalMusicModule.kt]: Lotfan esm pishnehad bedid

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt (hunk 14 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: yani adapter ro az Observer biaram birun?
ya faghat val music = it bezaram

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt (hunk 10 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: bale

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt (hunk 13 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: ye extension delegate e  be name:
`viewModels`
berid dakhele safehe doc ViewModel mibinid

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt (hunk 14 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: adapter hatman bayad biad biron

File: app/build.gradle.kts [hunk 15]
```diff
@@ -1,3 +1,5 @@
+import AndroidKts.fragmentKts
```
Comment (helloseyedjafari) [inline app/build.gradle.kts:1]: fek mikonam ezafie

File: app/src/main/java/com/worldsnas/starplayer/di/MusicProviderModule.kt [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+
+    @JvmStatic
+    @Provides
+    fun provideContentResolver(context: Context): ContentResolver {
+        return context.contentResolver
+    }
+
+    @JvmStatic
+    @Provides
+    fun provideLocalMusicImpl(application: Application): LocalMusicProvider {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/di/MusicProviderModule.kt:17]: chera az hamoni ke on bala sakhti estefade nakardi pas?

File: app/src/main/java/com/worldsnas/starplayer/di/MusicProviderModule.kt [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+import com.worldsnas.starplayer.model.LocalMusicProviderImpl
+import dagger.Module
+import dagger.Provides
+
+@Module
+object MusicProviderModule {
+
+    @JvmStatic
+    @Provides
+    fun provideContentResolver(context: Context): ContentResolver {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/di/MusicProviderModule.kt]: bejaye context `Application` bashe behtare

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    fun postMusic(): LiveData<List<Music>> {
+        getAllMusic()
+        return postMusicList
+    }
+
+    override fun onCleared() {
+        super.onCleared()
+        viewModelScope.cancel()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: be in niazi nist
khode ViewModel anjamesh mide
behtare kole function ro hazf konid

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 19]
```diff
@@ -24,6 +24,11 @@ class MusicsListAdapter(private val musics: ArrayList<Music>) :
     override fun onBindViewHolder(holder: BaseViewHolder<Music>, position: Int) {
         holder.onBind(musics[position])
     }
+
+    fun submitList(musicList: List<Music>) {
+        musics += musicList
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: in list nabayad toye constructor gerefte beshe
baye ye parameter khode class bashe
vali in taghyir e dorostie va khobe Afarin

Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: `notifyDataSetChanged` kheili function sanginie va tosie mishe azash dori beshe
sade tarin rah estefade az `ListAdapter<T, V>` hast bejaye class Adapter mamoli

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt [hunk 20]
```diff
@@ -1,23 +1,45 @@
...
+import javax.inject.Inject
+
 
 class MusicsListFragment : Fragment() {
 
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory
+
+    private val musicListViewModel by
+    viewModels<MusicListViewModel> { viewModelFactory }
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt:31]: afarin

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt [hunk 21]
```diff
@@ -26,31 +48,89 @@ class MusicsListFragment : Fragment() {
...
 
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
 
+    private fun liveDataSetup() {
+
+        val musicObserver = Observer<List<Music>> {
+            musics = it as ArrayList<Music>
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: cast kardan be in shekr kare khobi nist

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
Well done.
lot's of improvements

File: app/build.gradle.kts (hunk 15 above)
Comment (IamHaddad) [reply app/build.gradle.kts:1]: درسته فراموش کردم پاکش کنم

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt (hunk 21 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: pas chetor anjam bedam?

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+    private fun getAllMusic() {
+        viewModelScope.launch {
+            val musicsList = localMusicProvider.getAllMusic()
+
+            postMusicList.postValue(musicsList)
+        }
+    }
+
+    fun postMusic(): LiveData<List<Music>> {
+        getAllMusic()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: man bodam ino to `init` mizashtam ke yebar bishtar seda zade nashe

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 23]
```diff
@@ -3,14 +3,18 @@ package com.worldsnas.starplayer.view.musics_list
...
 import com.worldsnas.starplayer.core.BaseViewHolder
 import com.worldsnas.starplayer.databinding.ItemMusicBinding
 import com.worldsnas.starplayer.model.Music
 
-class MusicsListAdapter(private val musics: ArrayList<Music>) :
-    RecyclerView.Adapter<BaseViewHolder<Music>>() {
+class MusicsListAdapter() :
+    ListAdapter<Music, MusicListItemViewHolder>(DiffCallback) {
 
+    private val musics = ArrayList<Music>()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: in dige nabayad inja bashe ke

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 24]
```diff
@@ -21,7 +25,7 @@ class MusicsListAdapter(private val musics: ArrayList<Music>) :
         return musics.size
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: inja ham niazi nist

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 25]
```diff
@@ -21,7 +25,7 @@ class MusicsListAdapter(private val musics: ArrayList<Music>) :
...
     }
 
-    override fun onBindViewHolder(holder: BaseViewHolder<Music>, position: Int) {
+    override fun onBindViewHolder(holder: MusicListItemViewHolder, position: Int) {
         holder.onBind(musics[position])
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: in ham taghyir mikone

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
just a bit further

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt (hunk 23 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: درسته حواسم نبوده

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 26]
```diff
@@ -1,32 +1,30 @@
...
-    override fun getItemCount(): Int {
-        return musics.size
+    override fun onBindViewHolder(holder: MusicListItemViewHolder, position: Int) {
+        val musicItem = getItem(position)
+        holder.onBind(musicItem)
     }
 
-    override fun onBindViewHolder(holder: MusicListItemViewHolder, position: Int) {
-        holder.onBind(musics[position])
+    override fun submitList(list: List<Music>?) {
```
Comment (IamHaddad) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: nemidonm in kare khobi hast ya na

**Review by IamHaddad — COMMENTED** [review]
estefade az submitList khode ListAdapter

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt [hunk 27]
```diff
@@ -0,0 +1,33 @@
...
+import com.worldsnas.starplayer.model.LocalMusicProvider
+import com.worldsnas.starplayer.model.LocalMusicProviderImpl
+import com.worldsnas.starplayer.model.Music
+import kotlinx.coroutines.*
+import javax.inject.Inject
+
+class MusicListViewModel @Inject constructor(private val localMusicProvider: LocalMusicProvider) :
+    ViewModel() {
+    init {
+        getAllMusic()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt:17]: afarin

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt (hunk 26 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: lotfan override nakonid
inja niazi nist sshoma kari konid

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 28]
```diff
@@ -33,4 +35,14 @@ class MusicListItemViewHolder(private val mBinding: ItemMusicBinding) :
         mBinding.txtMusicTitle.text = obj.title
         mBinding.txtArtist.text = obj.artist
     }
+}
+
+object DiffCallback : DiffUtil.ItemCallback<Music>() {
+    override fun areItemsTheSame(oldItem: Music, newItem: Music): Boolean {
+        return oldItem == newItem
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: inja behtare `id` barresi beshe hamon tor ke az esmesh peydas

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 29]
```diff
@@ -33,4 +35,14 @@ class MusicListItemViewHolder(private val mBinding: ItemMusicBinding) :
...
+    }
+
+    override fun areContentsTheSame(oldItem: Music, newItem: Music): Boolean {
+        return oldItem.id == newItem.id
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: inja behtare tamame parameter ha `==` check beshe
ba tavajoh be esmesh

tip: piade sazi in do ro bar aks konid

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 30]
```diff
@@ -1,28 +1,30 @@
...
-        return musics.size
+    override fun onBindViewHolder(holder: MusicListItemViewHolder, position: Int) {
+        val musicItem = getItem(position)
+        holder.onBind(musicItem)
     }
 
-    override fun onBindViewHolder(holder: BaseViewHolder<Music>, position: Int) {
-        holder.onBind(musics[position])
+    override fun submitList(list: List<Music>?) {
+        super.submitList(list?.let { ArrayList(it) })
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: agar niaz be inkar has to laye ye ViewModel anjam bedid

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
Awesome.
you are getting a hang of it!

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt (hunk 29 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: behtar nist to model hash dorost konam , baad hash ro check konam?

**Review by helloseyedjafari — APPROVED** [review]
well done

**Pull request merged**